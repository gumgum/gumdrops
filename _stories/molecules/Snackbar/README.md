## Snackbar

The `<Snackbar>` component is a style wrapper for the `<SnackbarNotification>` components.

## Snackbar Notification

The `<SnackbarNotification>` component displays a notification inside the `<Snackbar>` component wrapper.
The notification can be hidden by clicking a button or after waiting for a timeout to complete.
A callback function can be executed after hiding a notification,
the first parameter of the callback will be the id of the notification if it is passed as a prop.

### Props

The following props may be passed to configure the `<SnackbarNotification>` component:

| name            | type              | description                                                                                                           | default |
| --------------- | ----------------- | --------------------------------------------------------------------------------------------------------------------- | ------- |
| id              | `String` `Number` | Unique key for identification, Required                                                                                         |         |
| text            | `String`          | Content of the notification, can be passed as children. If no text or children is passed, the component won't render. |         |
| context         | `String`          | Context of notification (success, info, warning, danger, hidden)                                                      | ``      |
| msToClose       | `Number`          | Time in ms to hide notification, if value is 0 the notification will remain open                                      | `3000`  |
| hideCloseButton | `Boolean`         | Hides the close button, if true auto enables timeout to hide notification                                             | `false` |
| callback        | `Function`        | Function to be called after hiding notification, passes notification id as first parameter when called                |         |
| className       | `String`          | Additional class to be added to the outermost element                                                                 |         |
| style           | `Object`          | Additional styles to be added to the outermost element                                                                | `false` |
