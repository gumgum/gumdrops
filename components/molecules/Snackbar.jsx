import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Snackbar = ({ children, className, style }) => {
    const baseClass = 'gds-snackbar';
    const rootClass = cx(baseClass, className);

    return (
        <ul className={rootClass} style={style}>
            {children}
        </ul>
    );
};

Snackbar.displayName = 'Snackbar';

Snackbar.propTypes = {
    /** Class names passed through props */
    className: PropTypes.string,
    /** Styles passed through props */
    style: PropTypes.object,
    /** Component children to use as notifications, use <SnackbarNotification> component */
    children: PropTypes.node
};

export default Snackbar;
