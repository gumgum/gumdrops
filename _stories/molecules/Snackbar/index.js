import React from 'react';
import { text, boolean, object, number } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { optionalSelect } from '../../../components/utils/optionalSelect';

import readme from './README.md';
import Snackbar from '../../../components/molecules/Snackbar';
import SnackbarNotification from '../../../components/atoms/SnackbarNotification';

const contextOptions = {
    success: 'success',
    info: 'info',
    warning: 'warning',
    danger: 'danger',
    hidden: 'hidden',
    'No Value': ''
};

const component = () => (
    <Snackbar>
        <SnackbarNotification
            id={text('ID', '1')}
            text={text(
                'Text',
                'You can test this notification options with the knobs on the right.'
            )}
            context={optionalSelect('Context', contextOptions, '')}
            msToClose={number('Milliseconds to hide', 0)}
            hideCloseButton={boolean('hideCloseButton', false)}
            callback={action('hide_callback')}
            className={text('ClassName', '')}
            style={object('Style', {})}
        />
        <SnackbarNotification
            id="2"
            text="This notification will hide in 5 secs..."
            context="danger"
            msToClose={5000}
            callback={action('autohide_callback')}
        />
        <SnackbarNotification id="3" msToClose={0} context="info">
            This notification content is passed as a child
        </SnackbarNotification>
    </Snackbar>
);

export default [readme, component];
