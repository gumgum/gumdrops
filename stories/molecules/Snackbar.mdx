import { Canvas, Source } from '@storybook/blocks';
import { ArgsTable } from '@storybook/addon-docs';

import * as SnackbarStories from './Snackbar.stories';
import Snackbar from '../../components/molecules/Snackbar';
import SnackbarNotification from '../../components/atoms/SnackbarNotification';

# Snackbar

The `<Snackbar>` component is a style wrapper for the `<SnackbarNotification>` components.

## Snackbar Notification

The `<SnackbarNotification>` component displays a notification inside the `<Snackbar>` component wrapper.

The notification can be hidden by clicking a button or after waiting for a timeout to complete.

A callback function can be executed after hiding a notification. The first parameter of the callback will be the id of the notification if it is passed as a prop.

## Component Example

<Canvas of={SnackbarStories.Default} />

## Component Example Source

<Source of={SnackbarStories.Default} />

## Snackbar Props

<ArgsTable of={Snackbar} />

## SnackbarNotification Props

<ArgsTable of={SnackbarNotification} />
