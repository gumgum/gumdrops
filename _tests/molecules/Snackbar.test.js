/* globals shallow */
import React from 'react';
import Snackbar from '../../components/molecules/Snackbar';

describe('Expect <Snackbar>', () => {
    it('to render', () => {
        const wrapper = shallow(<Snackbar />);
        expect(wrapper.find('.gds-snackbar').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });

    it('to render a child component', () => {
        const wrapper = shallow(
            <Snackbar>
                <li className="gds-snackbar__notification">
                    <p>Test</p>
                </li>
            </Snackbar>
        );
        expect(wrapper.find('.gds-snackbar__notification').length).toBe(1);
        expect(wrapper.find('p').text()).toMatch('Test');
        expect(wrapper).toMatchSnapshot();
    });
});
