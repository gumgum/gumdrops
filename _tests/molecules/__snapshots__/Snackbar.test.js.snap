// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Snackbar> to render 1`] = `
<ul
  className="gds-snackbar"
/>
`;

exports[`Expect <Snackbar> to render a child component 1`] = `
<ul
  className="gds-snackbar"
>
  <li
    className="gds-snackbar__notification"
  >
    <p>
      Test
    </p>
  </li>
</ul>
`;
